test/integration/client/cancel-query-tests.js

```diff
@@ -1,5 +1,20 @@
 var helper = require(__dirname+"/test-helper");
 
+test("cancellation of an active query", function() {
+  var client = helper.client();
+
+  var qry = "select pg_sleep(4)";
+  var query1 = client.query(qry);
+  client.cancel(client, query1);
+  var otherClient = helper.client();
```

Comment:
Postgres only allows for & handles one executing query per connection (client) at a time.  If you try to submit two on the same client the 2nd one will return an error message.  That's been a built in constraint of postgres for as long as I've been using it & how it behaves when consumed from any programming language.  It actually makes some sense when you think about it too...how would transactions work in a semi-non deterministic environment of sending 2 queries over the network at the same time working within a single transaction?  Yikes.  So to work with this constraint node-postgres does two things:

1) Provides a client pool so you can run multiple queries "at once" because you're sending them on multiple different clients.
2) The node-postgres client internally queues up pending queries so if you send two at once it keeps them ordered and dispatches them one after another.

So its a pain to cancel a query because you have to issue one query from one client & then connect & cancel it from another client - there's no real way around it.  The client that's busy waiting on a query wont send any data to the backend, and even if you *did* the backend would ignore any data coming in on the client which is currently processing a query and simply return an error on it.